src/python/gem5/components/processors/linear_generator.py

```diff
@@ -64,7 +64,7 @@ def __init__(
         generator cores that could replace the processing cores in a board.
 
         :param num_cores: The number of linear generator cores to create.
-        :param duration: The number of ticks for the generator to generate
+        :param duration: The duration of time for which the generator generates
```

Comment:
Is there anything you can reference so somneone reading this knows the format of the string to pass? Reading this my initial feeling is "what's the unit here?" but the parameter is a string so I assume it's "<integer><SI unit>s"?

Maybe a bit nitpicky, but would be nice to be specific.

((not in the scope for this PR but I think having duration in ticks is much better. That or an int or float for the number of, say, nano seconds to simulate. I really think a string is a terrible type for duration...))

Reply:
Hi @BobbyRBruce, I just updated the documentation for the duration parameter.

I also added some documentation for the data_limit parameter.

Let me know if there's anything else you'd like to see!

(I also agree that having an int or float for duration might be more intuitive.)

Thank you!